rcl/test/rcl/test_publisher.cpp

```diff
@@ -399,6 +436,14 @@ TEST_F(CLASSNAME(TestPublisherFixture, RMW_IMPLEMENTATION), test_invalid_publish
   rcl_reset_error();
   publisher.impl->context = saved_context;
 
+  // nullptr arguments
+  EXPECT_EQ(RCL_RET_INVALID_ARGUMENT, rcl_publish(&publisher, nullptr, nullptr));
```

Comment:
For these tests, it would be good to check that all variables are good except one, that way you know you're catching the failure case.

Reply:
For the `publish` functions, the last parameter is often used (at least in the tests) as a valid `nullptr`, any ideas on what to do in this case?

Reply:
I thought that might be the case, and so you might just clarify that the second nullptr is a valid argument. You could either add more to your comment, or declare a variable that communicates the same thing, a la:
```
rmw_publisher_allocation_t * null_allocation_is_valid_arg = nullptr;
```

Reply:
Good idea, I used it in old tests as well, see ac2f6a1